### PR TITLE
LPS-201728 ReindexSingleIndexerBackgroundTaskExecutor don't need to b…

### PR DESCRIPTION
**Pull request by rcsdnj**
…e @Component; BackgroundTaskExecutorConfigurator already registers it to the service registry

### CODE REVIEW

**Comment by rcsdnj** [comment]
ci:help

**Comment by rcsdnj** [comment]
ci:test:search

File: modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/background/task/ReindexSingleIndexerBackgroundTaskExecutor.java [hunk 1]
```diff
@@ -26,37 +25,31 @@
...
 public class ReindexSingleIndexerBackgroundTaskExecutor
 	extends BaseReindexBackgroundTaskExecutor {
 
+	public ReindexSingleIndexerBackgroundTaskExecutor(
+		IndexerRegistry indexerRegistry,
+		ReindexStatusMessageSender reindexStatusMessageSender,
+		SearchEngineHelper searchEngineHelper,
+		ServiceTrackerList<Indexer<?>> systemIndexers) {
+
+		this.indexerRegistry = indexerRegistry;
```
Comment (dantewang) [inline modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/background/task/ReindexSingleIndexerBackgroundTaskExecutor.java:42]: @rcsdnj After adding constructor, these fields no longer need to be protected, as there's no class extending this.

File: modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/background/task/ReindexSingleIndexerBackgroundTaskExecutor.java [hunk 2]
```diff
@@ -26,37 +25,31 @@
...
 	}
 
-	@Activate
-	protected void activate(BundleContext bundleContext) {
-		systemIndexers = ServiceTrackerListFactory.open(
-			bundleContext, (Class<Indexer<?>>)(Class<?>)Indexer.class,
-			"(system.index=true)");
-	}
-
 	@Deactivate
```
Comment (dantewang) [inline modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/background/task/ReindexSingleIndexerBackgroundTaskExecutor.java:53]: @rcsdnj After making it non-component, this method should also be removed.

File: modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/background/task/BackgroundTaskExecutorConfigurator.java [hunk 3]
```diff
@@ -35,22 +41,46 @@ protected void activate(BundleContext bundleContext) {
 				bundleContext, _concurrentReindexManagerSnapshot.get(),
 				_portalExecutorManager, _syncReindexManagerSnapshot.get());
 
+		systemIndexers = ServiceTrackerListFactory.open(
```
Comment (dantewang) [inline modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/background/task/BackgroundTaskExecutorConfigurator.java:44]: @rcsdnj Please keep `systemIndexers` in `ReindexSingleIndexerBackgroundTaskExecutor`, and expose a method to call `close()` on it.

In `deactivate()` method of this class, after unregistering, call that method.

**Review by dantewang — CHANGES_REQUESTED** [review]
cc @tinatian

**Comment by dantewang** [comment]
@rcsdnj Additionally, please include something describing the task like in other PRs (e.g. https://github.com/liferay-core-infra/liferay-portal/pull/6147).

**Comment by rcsdnj** [comment]
@dantewang, sorry, this PR was intended for internal tests and I've opened on liferay-core-infra by mistake. Anyway, I appreciate your comments and I'll do the requested changes before creating the "real" PR

**Pull request closed without merging**